applications/nrf_desktop/configuration/nrf52840dongle_nrf52840/app.overlay

```diff
@@ -28,6 +28,16 @@
 	};
 };
 
+&usbd {
+	compatible = "nordic,nrf-usbd";
+	status = "okay";
+	num-bidir-endpoints = <0>;
+	num-in-endpoints = <7>;
```

Comment:
The common configuration had to handle the worst possible use-case (with 4 USB HID class instances).

Together with changing approach, it would be reasonable to tailor the number of endpoints to needs of a given build type: https://github.com/nrfconnect/sdk-nrf/blob/main/applications/nrf_desktop/src/modules/usb_state.c#L125. We should also take into account additional USB endpoints used by the BLE QoS's USB CDC ACM class (to make it work without need to updating DTS manually).

Reply:
This was just a restructure of the DTS definitions. Your suggestion changes the DTS descriptions for each build type, which requires runtime tests. We can consider creating a follow-up ticket to optimize the USB part of DTS.

Reply:
A follow-up ticket should be ok here :)

Reply:
https://nordicsemi.atlassian.net/browse/NCSDK-30853